agent.go

```diff
@@ -191,6 +200,35 @@ func (a *Agent) register() error {
 	return nil
 }
 
+// runMetrics is a long-running goroutine that exposes an http metrics interface
+func (a *Agent) runMetrics() {
+	if a.config.Telemetry.PrometheusRetentionTime < 1 || a.config.MetricsInterface != "" {
+		return
+	}
+
+	mux := http.NewServeMux()
+	srv := &http.Server{Addr: a.config.MetricsInterface, Handler: mux}
+	handlerOptions := promhttp.HandlerOpts{
+		ErrorLog:      a.logger,
+		ErrorHandling: promhttp.ContinueOnError,
+	}
+
+	mux.Handle("/metrics", promhttp.HandlerFor(prometheus.DefaultGatherer, handlerOptions))
+	go func() {
+		<-a.shutdownCh
+		deadline := time.Now().Add(5 * time.Second)
+		ctx, cancel := context.WithDeadline(context.Background(), deadline)
+		defer cancel()
+		if err := srv.Shutdown(ctx); err != nil {
+			a.logger.Printf("[ERR] Failed to shutdown metrics interface: %v", err)
+		}
+	}()
+
+	if err := srv.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+		a.logger.Printf("[ERR] Failed to open metrics interface: %v", err)
```

Comment:
It's the other way around. If the error is NOT `ErrServerClosed` it is logged, otherwise nothing is printed because we are just shutting down normally.

I did not see other components bringing down ESM when they encountered an error (most retried), and since this is just a metrics interface for now, it seemed better to just log and continue. But I'm open to suggestions here, how should I bring ESM down? Write to `shutdownCh`?

Reply:
Yes! You're right, the error is logged when it is not `ErrServerClosed `, which sounds good to me.

I agree, most components don't stop ESM. If you prefer to log and continue, that works for me. I was mostly thinking about what would be most helpful for you and the community. It sounds like your preference is log and continue, so let's go with that.